sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -641,12 +641,21 @@ bool ProgramManager::isSpecialDeviceImageShouldBeUsed(
     // TODO: re-design the encode of the devicelib metadata if we must support
     // more devicelib images in this way.
     enum { DEVICELIB_FALLBACK = 0, DEVICELIB_NATIVE };
+    ur_bool_t NativeBF16Supported = false;
     const std::shared_ptr<detail::device_impl> &DeviceImpl =
         detail::getSyclObjImpl(Dev);
-    std::string NativeBF16ExtName = "cl_intel_bfloat16_conversions";
-    bool NativeBF16Supported = (DeviceImpl->has_extension(NativeBF16ExtName));
-    return NativeBF16Supported ==
-           (Bfloat16DeviceLibVersion == DEVICELIB_NATIVE);
+    bool CallSuccessful =
+        DeviceImpl->getAdapter()->call_nocheck<UrApiKind::urDeviceGetInfo>(
+            DeviceImpl->getHandleRef(),
+            UR_DEVICE_INFO_BFLOAT16_CONVERSIONS_NATIVE, sizeof(ur_bool_t),
+            &NativeBF16Supported, nullptr);
+    if (!CallSuccessful) {
```

Comment:
Hi, @steffenlarsen 
It makes sense to check whether the urDeviceGetInfo call succeeds, I explicitly check the result to make the logic easier to understand and add some comments.

Thanks very much.

Reply:
Thanks!